### PR TITLE
Unify industrial and github argument syntax

### PR DESCRIPTION
**Pull request by rfeistenauer**
## Description

makes the commands for github runner and industrial runner interchangable.

e.g. 
```
rosrun industrial_ci run_ci ROS_DISTRO=noetic ROS_REPO=main DOCKER_RUN_OPTS="-v /usr/local/share/ca-certificates:/usr/local/share/ca-certificates:ro" APT_PROXY="http://172.20.20.104:3142"
```
becomes 
```
rosrun pilz_github_ci_runner test_repository.py "PilzDE/psen_scan_v2" "rfeistenauer agutenkunst" ROS_DISTRO=noetic ROS_REPO=main DOCKER_RUN_OPTS="-v /usr/local/share/ca-certificates:/usr/local/share/ca-certificates:ro" APT_PROXY="http://172.20.20.104:3142"
```

### CODE REVIEW

**Review by agutenkunst — CHANGES_REQUESTED** [review]
see inline comments

File: scripts/test_repository.py [hunk 1]
```diff
@@ -49,6 +45,7 @@
 import os
 import sys
 import time
+import shlex
```
[the remark this answers is not in the document]
Comment (rfeistenauer) [reply scripts/test_repository.py:48]: I use it in line 123. It may not be strictly neccesary there but its usually the better option to bare split() calls if you are handling shell commands as far as I know. Is there a special reason you want to get rid of it?

File: src/pilz_github_ci_runner/hardware_tester.py [hunk 2]
```diff
@@ -21,17 +21,24 @@
...
         self._log_dir = log_dir
-        self._docker_opts = docker_opts
-        self._cmake_args = cmake_args
-        self._apt_proxy = apt_proxy
+        self._env = self._create_ci_env(ci_args)
         self._setup_cmd = setup_cmd
         self._cleanup_cmd = cleanup_cmd
 
+    def _create_ci_env(self, ci_args):
+        relevant_env = {k: os.environ.get(k) for k in KEYS if k in os.environ}
```
[the remark this answers is not in the document]
Comment (rfeistenauer) [reply src/pilz_github_ci_runner/hardware_tester.py]: "Is it necessary that the variables are actually os environment variables?"
No, It is not neccessary for them to be actually os env variables it will take:
 - system environment vriables
 - variables defined in front of the command aka ROS_DISTRO=melodic rosrun pilz_github...
 - variables passed as CI_ARGS

priority is: CI_ARGS > in fron of command > system environment

"What is the reasoning behind these lines?"
Added another function and renamed this one to hopefully clarify the idea behind this lines. 

For what {k: os.environ.get(k) for k in KEYS if k in os.environ} does in special lookup dict comprehension in python. 
In short: It loops through the KEYS and creates a new dict which it fills with all the os.env variables having such a key and exist.

File: scripts/test_repository.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (agutenkunst) [reply scripts/test_repository.py:48]: Since `shlex` is not part of https://github.com/ros/rosdistro/blob/master/rosdep/python.yaml I would drop it.

[the remark this answers is not in the document]
Comment (rfeistenauer) [reply scripts/test_repository.py:48]: But it is standard library so it should not be required to be in rosdep
see https://docs.python.org/3.9/library/index.html
